lib/modules.nix

```diff
@@ -223,7 +223,12 @@ rec {
           };
         in parentFile: parentKey: initialModules: args: collectResults (imap1 (n: x:
           let
-            module = loadModule args parentFile "${parentKey}:anon-${toString n}" x;
+            defs = [{ file = parentFile; value = x; }];
+            module =
+              if builtins.isList x then
```

Comment:
Do this check inside `loadModule` instead

Reply:
Fixed